language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/TransactionsServiceImpl.scala

```diff
@@ -66,61 +71,50 @@ class TransactionServiceImpl(ledgerContent: Observable[LedgerItem]) extends Tran
       request: GetTransactionsRequest,
       responseObserver: StreamObserver[GetTransactionTreesResponse]): Unit = {
     lastTransactionsTreesRequest.set(request)
-//  TODO DEL-6007
-//    ledgerContent
-//      .map(t => GetTransactionTreesResponse(List(t.toTransactionTree)))
-//      .foreach(responseObserver.onNext)
-
     responseObserver.onCompleted()
   }
 
   override def getTransactionByEventId(
-      request: GetTransactionByEventIdRequest): Future[GetTransactionResponse] =
-    Future.failed[GetTransactionResponse] {
-      lastTransactionByEventIdRequest.set(request)
-      //  TODO DEL-6007
-      new StatusRuntimeException(Status.UNIMPLEMENTED.withDescription("Pending"))
-    }
+      request: GetTransactionByEventIdRequest): Future[GetTransactionResponse] = {
+    lastTransactionByEventIdRequest.set(request)
+    Future.successful(new GetTransactionResponse(None)) // just a mock, not intended for consumption
```

Comment:
What does this comment mean? Isn't this simply an empty response?
This is definitely not a mock as it does not record calls (according to nomenclature e.g. https://blog.pragmatists.com/test-doubles-fakes-mocks-and-stubs-1a7491dfa3da).

Reply:
`lastTransactionByEventIdRequest.set(request)` - it records it in the line above, no?

Reply:
Isn't that what the line above this one does?
```scala
    lastTransactionByEventIdRequest.set(request)
```

Reply:
You're right -- this is indeed a mock.